libsolidity/formal/SMTEncoder.cpp

```diff
@@ -891,6 +891,15 @@ bool SMTEncoder::visit(MemberAccess const& _memberAccess)
 				IntegerType const& integerType = dynamic_cast<IntegerType const&>(*magicType->typeArgument());
 				defineExpr(_memberAccess, memberName == "min" ? integerType.minValue() : integerType.maxValue());
 			}
+			else if (memberName == "interfaceId")
+			{
+				// TODO: move this calculation into ContractDefinition and share with ExpressionCompiler
+				ContractDefinition const& contract = dynamic_cast<ContractType const&>(*magicType->typeArgument()).contractDefinition();
```

Comment:
```suggestion
				auto const& contract = dynamic_cast<ContractType const&>(*magicType->typeArgument()).contractDefinition();
```

Reply:
I actually think the type may be useful here because `contractDefinition` is not clear, e.g. the type does match with `dynamic_cast`.

Also both this and the one above for `IntegerType` was copied from `ExpressionCompiler.